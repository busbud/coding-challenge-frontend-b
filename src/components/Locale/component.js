import React from 'react';
import { string, func } from 'prop-types';
import { MenuItem } from 'material-ui/Menu';

const propTypes = {
  code: string.isRequired,
  onClick: func.isRequired,
};

const Component = ({ code, onClick }) => (
  <MenuItem key={code} onClick={() => onClick(code)}>
    {code}
  </MenuItem>
);

Component.propTypes = propTypes;

export default Component;
