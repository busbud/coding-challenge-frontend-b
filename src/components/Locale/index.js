import React from 'react'
import { connect } from 'react-redux'

import * as localesSelectors from '../../store/services/locales/selectors'

import Container from '../Container'

import {
  StyledLocale,
} from './styles'

const Locale = (props) => (
  <StyledLocale>
    <Container>
      <select
        onChange={props.changeLocale}
        value={props.lang}>
        {props.locales.map(locale => (
          <option key={locale.lang} value={locale.lang}>{locale.lang.toUpperCase()} - {locale.currency}</option>
        ))}
      </select>
    </Container>
  </StyledLocale>
)

const mapStateToProps = state => {
  return {
    lang: localesSelectors.getLang(state),
  }
}

export default connect(mapStateToProps)(Locale)