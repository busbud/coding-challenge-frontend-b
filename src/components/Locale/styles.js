import styled from 'styled-components'

export const StyledLocale = styled.div`
  background-color: #0b72be;
  font-size: 14px;
  text-align: right;
  padding: 1.42rem 1.14rem;
  position: relative;
`
