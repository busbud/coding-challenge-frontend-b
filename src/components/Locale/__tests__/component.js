import React from 'react';
import { shallow, render } from 'enzyme';
import Locale from '../component';

const props = {
  code: 'en',
  onClick: jest.fn(),
};

describe('component | Locale | component', () => {
  describe('render', () => {
    it('should render without throwing and error', () => {
      // given
      const wrapper = shallow(<Locale {...props} />);

      // then
      expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
      // given
      const wrapper = shallow(<Locale {...props} />);

      // then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Locale', () => {
    it('should have proper key', () => {
      // given
      const wrapper = shallow(<Locale {...props} />);
      const key = wrapper.key();

      // then
      expect(key).toBe(props.code);
    });
    it('should have proper innerText', () => {
      // given
      const wrapper = render(<Locale {...props} />);
      const text = wrapper.text();

      // then
      expect(text).toBe(props.code);
    });
    it('should call onClick prop function with proper params', () => {
      // given
      const wrapper = shallow(<Locale {...props} />);

      // when
      wrapper.simulate('click');

      // then
      expect(props.onClick).toHaveBeenCalledWith(props.code);
    });
  });
});
