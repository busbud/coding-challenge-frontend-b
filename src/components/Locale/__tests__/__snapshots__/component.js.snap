// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component | Locale | component render should match snapshot 1`] = `
<WithStyles(MenuItem)
  key="en"
  onClick={[Function]}
>
  en
</WithStyles(MenuItem)>
`;
